### PR TITLE
Add event tracking

### PR DESCRIPTION
**Pull request by kr8n3r**
## What 

To measure engagement for users that have accepted cookies we want to
record certain events with either specific labels for targeted elements
or more generic for larger groups .

This script introduces functions to send events for:

- targeted elements with data attributes
- external links to non `*.cloud.service.gov.uk` subdomains and email links
- 404 pages

For targeted elements values for the event are read from the data
attributes.

For external links, values are partially set and partially read from
the target element.

For 404 pages, values are set.

## How to review
- Test url:  https://paas-product-pages-ga-test.london.cloudapps.digital/ (being verified by PA)
or  locally: 
- clone branch
- run `npm install` to install latest dependencies
- run `npm test` to verify
- review code

### CODE REVIEW

**Review by 0atman — CHANGES_REQUESTED** [review]
Just a few questions about this, maybe I've been out of the js game too long, please accept my apologies if I've got the wrong idea.

I find it very strange that analytics.js has inconsistent semicolon use. Personally, I don't use them in my js projects (I use https://standardjs.com to warn me in the very few cases where they're needed, but maybe we use something else in paas?) is this an oversight or to a standard?

Thanks!

File: javascripts/analytics.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  if (this.is404Page) {
+    this.page404Event();
+  }
+}
+
+Analytics.prototype.handleClickEvent = function (e) {
+  var target = e.target,
+      isLink = e.target.nodeName === 'A',
+      // external link: starts with http(s) or mailto and does not contain a "service.gov" in url
+      externalLinkRegex = /^(?=.*(http(s)|mailto))(?:(?!cloud\.service).)*$/g,
```
[the remark this answers is not in the document]
Comment (0atman) [reply javascripts/analytics.js:19]: screenshot from https://regexr.com/

File: javascripts/analytics.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+  eventParams.event_category = 'Error'
+  eventParams.event_label = 'Page ' + prettyRequestedPageName
+  
+  this.sendEvent(action, eventParams)
+}
+
+Analytics.prototype.sendEvent = function (action, options) {
+  window.dataLayer = window.dataLayer || []
+  function gtag(){
```
[the remark this answers is not in the document]
Comment (kr8n3r) [reply javascripts/analytics.js:83]: was going by https://developers.google.com/analytics/devguides/collection/gtagjs but certainly can update that

File: javascripts/analytics.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (kr8n3r) [reply javascripts/analytics.js:19]: re: standardJS we currently don't have it setup in this project, paas-admin uses `eslint`

I'll remove all semicolons in here to make it compatible down the line

[the remark this answers is not in the document]
Comment (kr8n3r) [reply javascripts/analytics.js]: updated

File: javascripts/analytics.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  if (this.is404Page) {
+    this.page404Event()
+  }
+}
+
+Analytics.prototype.handleClickEvent = function (e) {
+  var target = e.target,
+      isLink = e.target.nodeName === 'A',
+      // external link: starts with http(s) or mailto and does not contain a "service.gov" in url
+      externalLinkRegex = /^(?=.*(http|mailto))(?:(?!cloud\.service\.gov\.uk).)*$/g,
```
Comment (0atman) [inline javascripts/analytics.js:19]: Now we're talking! Looks good to me :ok_hand: 

Screenshot from 2021-04-28 14-50-07

**Comment by kr8n3r** [comment]
@0atman  added a commit just now :)

**Comment by kr8n3r** [comment]
need to drop the standard fix commit

**Review by 0atman — APPROVED** [review]

**Pull request merged**